var/spack/repos/builtin/packages/mpifileutils/package.py

```diff
@@ -38,7 +39,17 @@ class Mpifileutils(Package):
     depends_on('dtcmp@1.0.3',  when='@:0.7')
     depends_on('dtcmp@1.1.0:', when='@0.8:')
 
+    # fixes were added to libarchive somewhere between 3.1.2 and 3.5.0
+    # which helps with file names that start with "._", bumping to newer
+    # libarchive, but in a way that does not disrupt older mpiFileUtils installs
     depends_on('libarchive')
+    depends_on('libarchive@3.5.1:', when='@0.11:')
+
+    depends_on('bzip2')
```

Comment:
Are these new dependencies applicable to all versions of the software that could be installed or a subset?

Reply:
The libarchive dependency only matters for versions 0.11 and later.

The bzip2 dependency really goes back through 0.8.